ng/src/web/components/folding/folding.js

```diff
@@ -107,12 +116,16 @@ export const withFolding = (Component, defaults = {}) => {
     }
 
     return (
-      <Component
-        style={new_style}
-        {...props}
-        onTransitionEnd={onFoldStepEnd}
-        onAnimationEnd={onFoldStepEnd}
-      />
+      <Div
+        {...new_style}
+      >
+        <Component
+          style={new_style}
+          {...props}
+          onTransitionEnd={onFoldStepEnd}
+          onAnimationEnd={onFoldStepEnd}
```

Comment:
Maybe both handlers can be added to the Div instead. But I am not sure about that. Maybe Component isn't generic and is always some defined component which requires both handlers.

Reply:
I tried moving them into the Div and the last remaining instance, where folding is used, still functions fine.